docs/docs/location-data-from-props.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Location Data from Props"
+---
+
+## What is location data
+
+Sometimes it can be helpful to know exactly what your app's browser URL is at any given stage. Using [@reach/router](https://github.com/reach/router) for [client-side routing](/docs/glossary#client-side) in Gatsby, `location` data represents where the app is currently, where you'd like it to go, and other helpful information. The `location` object is never mutated but `reach@router` makes it helpful to determine when navigation happens.
+
+```js
+{
+  key: 'ac3df4', // does not populate with a HashHistory!
+  pathname: '/somepage',
+  search: '?someurlquery=searching-string',
+  hash: '#about',
+  state: {
+    [userDefined]: true
+  }
+}
+```
+
+## HashHistory
+
+Using `hash` is one way to update the browser URL and the DOM without having the browser do a full reload. The HashHistory is used to track browser history when using [hashrouter](https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/HashRouter.md#hashrouter) instead of [browserrouter](https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/BrowserRouter.md#browserrouter) which uses the newer HTML5 history API.
+
+## Use cases
+
+Through client-side routing in Gatsby you can provide a location object instead of strings, helpful in a number of situations.
```

Comment:
```suggestion
Through client-side routing in Gatsby you can provide a location object instead of strings, which are helpful in a number of situations:
```